build.md

```diff
@@ -33,22 +42,31 @@ Here's a description of the various artifacts as you would see them in a release
 
 * The Debian package for Ubuntu-like Linux (to be `apt-get install`ed)
 
-* Some other stuff that's always there
-
 ### Running the CLI
 So, you built from source and you would like to run it. 
 
 * The first thing you need to do is to extract the binary for your OS from the `./package` directory. Un-tar the file that matches your OS.
 
 * Next, you need to copy that file to some place that's in your $PATH (for example, /usr/bin or /usr/local/bin) or %PATH% (any Win folder, then add it to the %PATH%). You may also want to call it `appsody`, so that you can run the CLI just by typing `appsody <command>`.
 
-* Next, you need to build the `appsody-controller` (sorry). It's here: https://github.com/appsody/controller. Just build the binary, and move it to the /usr/local/Cellar/appsody/`<release_tag>`/appsody-controller. Call it `appsody-controller` (mandatory!) and make sure that it has +x permissions (also mandatory).
-
 * If you are replacing an old installation, you may want to delete the appsody home directory (`$HOME/.appsody`).
 
+### The Appsody Controller
+In order to enable rapid app development, the Appsody CLI relies on a component called the `Appsody controller`, which is built separately, and which is meant to be run within the container that hosts the Appsody app itself (as opposed to being executed on the developers' system). However, the good news is that the Appsody CLI automatically downloads the Appsody controller when necessary, in the form of a Docker image.
+
+When the Appsody CLI is built using the `Makefile`, the CLI "knows" which version of the controller needs to be obtained. If you just compile the Appsody CLI binary using `go build`, the Appsody CLI will pull the `latest` version of the controller image (which may or may not be a good match - so, be aware of that).
+
+As a developer, you can also override which version of the Appsody controller the CLI uses, in two different ways:
+1) By setting the `APPSODY_CONTROLLER_VERSION` env var prior to launching the Appsody CLI. For example, `export APPSODY_CONTROLLER_VERSION=0.3.0`
```

Comment:
Are we deliberately not documenting these within the env vars on the website...?

Reply:
I assume you refer to these: https://appsody.dev/docs/stacks/environment-variables
Well... those are different env vars, in that they are consumed by the stacks. The env vars described here (and there are others, which we haven't documented since they cater to very specific audiences like Codewind) are consumed by the the CLI. I'd keep them separate - definitely we don't want to document in the same place as the stack env vars.
We had some discussions on how, where, and if we should document the CLI env vars...